### PR TITLE
[WIP] docs: add react isomorphic doc

### PR DESCRIPTION
**Pull request by jtyjty99999**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: docs/source/zh-cn/tutorials/react.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+- 数据来源与数据同步
+
+前端的数据应该与服务端保持一致，那么前端如何得到服务端的数据呢？很简单，页面渲染完毕之后，把数据挂载到全局对象上即可。
+
+- 服务器端渲染与前端渲染
+- 路由
+- 消除运行环境的差异
+
+我们可以通过 Webpack 加载器将样式、图片等前端资源作为模块进行引入，但这些资源如果在 Node 中直接引入，是会报错的。
```
Comment (atian25) [inline docs/source/zh-cn/tutorials/react.md]: - 官方貌似是 `webpack` 小写?
- `NodeJS` && `Node`  -> `Node.js`

File: docs/source/zh-cn/tutorials/react.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+另外一些浏览器端的 api 也需要 hack 掉。
+
+
+## 技术实现
+
+我们这次进行的同构，选型采用了 egg + egg-react插件 + React + Redux + React-Router + Webpack;
```
Comment (atian25) [inline docs/source/zh-cn/tutorials/react.md]: `[egg-react]` 加上链接

File: docs/source/zh-cn/tutorials/react.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+由于服务端 NodeJS 环境的存在，维护一套业务代码，可以分别在服务器端和前端运行。
+
+## 为什么要前后端同构
+
+React 前后端同构主要是相对于纯 React 浏览器渲染来比较的。
+
+- 更利于 SEO
+- 首屏加载更快
+- 提高了复用性，业务代码易于维护
```
Comment (dead-horse) [inline docs/source/zh-cn/tutorials/react.md]: 相对于纯浏览器渲染，没法提高复用性和维护性吧

File: docs/source/zh-cn/tutorials/react.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+title: 实现 React 同构渲染
+---
+
+什么是前后端同构（isomorphic）？同构可以理解为同一个组件或逻辑只编写一次，前后端可以共用。
```
Comment (popomore) [inline docs/source/zh-cn/tutorials/react.md]: 真的要将同构么，感觉我们也没最佳实践

Comment (atian25) [inline docs/source/zh-cn/tutorials/react.md]: 先写完再看看？到时也邀请达峰碰撞下

Comment (jtyjty99999) [inline docs/source/zh-cn/tutorials/react.md]: @popomore @atian25 算是抛砖引玉了吧

File: docs/source/zh-cn/tutorials/react.md (hunk 1 above)
Comment (huacnlee) [inline docs/source/zh-cn/tutorials/react.md]: webpack
Node.js

File: docs/source/zh-cn/tutorials/react.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+由于服务端 NodeJS 环境的存在，维护一套业务代码，可以分别在服务器端和前端运行。
```
Comment (fengmk2) [inline docs/source/zh-cn/tutorials/react.md]: Node.js 才是对的，NodeJS 是山寨写法。。。

File: docs/source/zh-cn/tutorials/react.md (hunk 3 above)
Comment (dead-horse) [inline docs/source/zh-cn/tutorials/react.md]: 如果要写这一章节，起码需要对比下面这三种的优劣。

1. 服务端模板引擎（Nunjucks） + 客户端 React（异构）
2. 服务端 React + 客户端 React（同构）
3. 纯客户端 React

Comment (jtyjty99999) [inline docs/source/zh-cn/tutorials/react.md]: @dead-horse 好的，WIP中

File: docs/source/zh-cn/tutorials/react.md (hunk 1 above)
Comment (okoala) [inline docs/source/zh-cn/tutorials/react.md]: 其实可以比较一些常用的同构渲染的方案，比如Steam渲染和String渲染的区别。包括针对渲染组件报错后的异常处理。

**Comment by popomore** [comment]
哈哈 @jtyjty99999 的坑太多了

File: docs/source/zh-cn/tutorials/react.md (hunk 1 above)
Comment (atian25) [inline docs/source/zh-cn/tutorials/react.md]: @okoala 有没有兴趣来写写~

**Comment by jtyjty99999** [comment]
@popomore 哎，慢慢填

**Comment by chengnuo** [comment]
egg-view-react 怎么用啊。加入配置
exports.view-react = {
^^^^^^^^^^^^^^^^^^

ReferenceError: Invalid left-hand side in assignment
    at createScript (vm.js:74:10)
    at Object.runInThisContext (vm.js:116:10)
    at Module._compile (module.js:533:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.loadFile (F:\myproject\X\x-time-management\node_modules\egg-core\lib\utils\index.js:18:19)
2017-07-17 13:44:14,043 ERROR 4412 nodejs.AgentWorkerDiedError: [master] agent_worker#1:25212 died (code: 1, signal: null)
    at Master.onAgentExit (F:\myproject\X\x-time-management\node_modules\egg-cluster\lib\master.js:274:17)
    at emitOne (events.js:115:13)
    at Master.emit (events.js:210:7)
    at Messenger.sendToMaster (F:\myproject\X\x-time-management\node_modules\egg-cluster\lib\utils\messenger.js:122:17)
    at Messenger.send (F:\myproject\X\x-time-management\node_modules\egg-cluster\lib\utils\messenger.js:87:12)
    at ChildProcess.agentWorker.once (F:\myproject\X\x-time-management\node_modules\egg-cluster\lib\master.js:181:22)
    at Object.onceWrapper (events.js:318:30)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
name: 'AgentWorkerDiedError'
pid: 4412
hostname: DESKTOP-273OIML

2017-07-17 13:44:14,043 ERROR 4412 [master] agent_worker#1:25212 start fail, exiting with code:1
2017-07-17 13:44:14,043 ERROR 4412 [master] exit with code:1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! x-time-management@1.0.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the x-time-management@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Administrator\AppData\Roaming\npm-cache\_logs\2017-07-17T05_44_14_056Z-debug.log

**Pull request closed without merging**